### ISSUE TITLE
Pivot fields of nova relations to the same resource in via the same pivot table 

### ISSUE THREAD

**Issue by buttjer** [laravel/nova-issues#2625]
- Laravel Version: 7.11
- Nova Version: 3.6
- PHP Version: 7.4
- Database Driver & Version: mySql
- Operating System and Version: Docker for Mac
- Browser type and version: Chrome 81

### Description:

If you have a Nova Resource which is related mutliple times to the same model, via the same pivot table. And two BelongsToMany Fields and different pivot fields. You will displayed always the first pivot fields.

You would expect to have the input fields of the specific fields method. Not always the first.
Both relation pivot forms are the same.

### Steps To Reproduce:

Nova/Product.php
```php

            BelongsToMany::make('Article', 'bundledArticles', Article::class)->fields(function () {
                    return [
                        Select::make('Type', 'type')->options([
                            'standard' => 'Standard',
                            'bundled' => 'Bundled,
                        ])->required()->rules('required'),
                        Text::make('goldlabel')
                    ];
                }),

            BelongsToMany::make('Article', 'articles', Article::class)->fields(function () {
                    return [
                        Select::make('Type', 'type')->options([
                            'standard' => 'Standard User',
                            'gold' => 'Gold User,
                        ])->required()->rules('required')
                    ];
                }),
```

Product.php
```php

    public function articles()
    {
        return $this->belongsToMany(Article::class)
            ->withPivot('type', 'quantity')->wherePivotIn('type', ['standard']);
    }

    public function bundledArticles()
    {
        return $this->belongsToMany(Article::class)
            ->withPivot('type', 'quantity')->wherePivotIn('type', ['bundled']);
    }

```

**Comment by buttjer** [comment]
Maybe related to #1745

**Comment by buttjer** [comment]
related to #2128

**Comment by SiebeVE** [comment]
This issues is actually a much bigger thing then written here.

Not only will it take the first in the pivot fields, but when you actually edit a pivot field, it will update all pivot fields of other records to.

This results in the loss of data of all pivot fields.

**Comment by sheggi** [comment]
+1

This is still the case with Nova 3.7.1

**Comment by f-liva** [comment]
Any way to get this fixed?

**Comment by SiebeVE** [comment]
I can't see an easy way to fix this. I've fixed it with adding an ID to the rows of the pivot table. Then using that ID in the views for editing, deleting,... It is a rather heavy customisation that won't fit everyone needs.

**Comment by IMPMAC** [comment]
I am also experiencing this issue. If you have multiple belongsToMany, the pivot fields of other fields show up in error.
The request to get the fields are a GET request to :
`/nova-api/products/creation-pivot-fields/products?editing=true&editMode=attach`

So it seems it does not specify the relationship you are trying to get the pivot fields for so it can distinguish what fields to show up.

**Comment by navidnadali** [comment]
> This issues is actually a much bigger thing then written here.
> 
> Not only will it take the first in the pivot fields, but when you actually edit a pivot field, it will update all pivot fields of other records to.
> 
> This results in the loss of data of all pivot fields.

Yeah this is pretty bad for our use case too as described above. Sorry @crynobone @davidhemphill is there any plans to look at a fix for this please?

**Comment by crynobone** [comment]
> This issues is actually a much bigger thing then written here.
    
 This perfectly describe the current state of the issue. I personally have try to look into this and it require major changes to the way pivot table is used in Laravel Nova.
 
 * We need to introduce `PivotID`, instead of just relying on `ID`. This would require eloquent relationship to use `Pivot` model explicitly, even then we need to have a way to verify that the pivot model contain a unique id. One way is using  `$incrementing = true;`.
 * We would need to change attaching pivot to allow multiple same relation with different pivot data.
 * We would need to allow editing individual pivot for above scenario while ensuring existing feature works fine.
 * This would also need to be handle for delete (and/or soft deletes).
 
 At the moment I can't guarantee a fixed date when this can be solve but we are looking into it and will release it once we confident with the solution.

**Comment by crynobone** [comment]
![image](https://user-images.githubusercontent.com/172966/108063934-eb4c8d00-7096-11eb-9ca9-613c5cec10f0.png)

PR submitted but still working on adding additional tests for it.

**Comment by crynobone** [comment]
Fixed and released with v3.23.1.
